### ISSUE TITLE
It would be nice to have .detach() and/or a gradient reversal function

### ISSUE THREAD

**Issue by HughRunyan** [facebookresearch/SparseConvNet#130]
So the SCN pieces can be used to build GAN-type networks without a lot of flipping back and forth between sparse and dense tensors.

**Comment by btgraham** [comment]
I have added '.detach()'.
However, building sparse GAN style networks (eg https://arxiv.org/abs/1811.10355) is non-trivial as the decoder can only teach the generator through active locations ...

**Comment by HughRunyan** [comment]
Thanks! Have you found that to be a big problem? I figured, like with standard supervised training, augmentations would eventually spread the information around. Seems likely I'm missing something.
